mmcv/runner/hooks/logger/pavi.py

```diff
@@ -120,24 +154,15 @@ def log(self, runner) -> None:
     @master_only
     def after_run(self, runner) -> None:
         if self.add_last_ckpt:
-            ckpt_path = osp.join(runner.work_dir, 'latest.pth')
-            if osp.islink(ckpt_path):
-                ckpt_path = osp.join(runner.work_dir, os.readlink(ckpt_path))
-
-            if osp.isfile(ckpt_path):
-                # runner.epoch += 1 has been done before `after_run`.
-                iteration = runner.epoch if self.by_epoch else runner.iter
-                return self.writer.add_snapshot_file(
-                    tag=self.run_name,
-                    snapshot_file_path=ckpt_path,
-                    iteration=iteration)
+            self._add_ckpt(runner, last_ckpt=True)
 
         # flush the buffer and send a task ending signal to Pavi
         self.writer.close()
 
     @master_only
     def before_epoch(self, runner) -> None:
-        if runner.epoch == 0 and self.add_graph:
+        if self.add_graph and ((runner.epoch - self.add_graph_start) %
```

Comment:
When `runner.epoch` is less than `add_graph_start` and `(runner.epoch - self.add_graph_start)` divisible interval, this will return true.